File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Solution {
    public string solution(string[] survey, int[] choices) {
        StringBuilder sb = new StringBuilder();
        var database = new Dictionary<char, int>()
        {
            {'R', 0}, {'T', 0},
            {'C', 0}, {'F', 0},
            {'J', 0}, {'M', 0},
            {'A', 0}, {'N', 0}
        };
        
        for(int i = 0; i < survey.Length; i++)
        {
            if(choices[i] <= 3)
            {
                database[survey[i][0]] += (4 - choices[i]);
            }
            else 
            {
                database[survey[i][1]] += (choices[i] - 4);
            }
        }
        if(database['R'] >= database['T'])
        {
            sb.Append('R');
        }
        else sb.Append('T');
        
        if(database['C'] >= database['F'])
        {
            sb.Append('C');
        }
        else sb.Append('F');
        
        if(database['J'] >= database['M'])
        {
            sb.Append('J');
        }
        else sb.Append('M');
        
        if(database['A'] >= database['N'])
        {
            sb.Append('A');
        }
        else sb.Append('N');
        
        return sb.ToString();
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    map<char, int> score;
    
    for(int i = 0; i < survey.size(); i++)
    {
        char a = survey[i][0];
        char b = survey[i][1];
        int choice = choices[i];
        
        if(choice > 4)
        {
            score[b] += (choice - 4);
        }
        else if(choice < 4)
        {
            score[a] += (4 - choice);
        }
    }
    
    answer += (score['R'] >= score['T']) ? 'R' : 'T';
    answer += (score['C'] >= score['F']) ? 'C' : 'F';
    answer += (score['J'] >= score['M']) ? 'J' : 'M';
    answer += (score['A'] >= score['N']) ? 'A' : 'N';
    
    return answer;
}